java-frontend/src/main/java/org/sonar/java/model/statement/SwitchExpressionTreeImpl.java

```diff
@@ -101,7 +101,7 @@ public void accept(TreeVisitor visitor) {
 
   @Override
   public Iterable<Tree> children() {
-    return Iterables.concat(
+    return ListUtils.concat(
```

Comment:
The more I read about all these changes, the more I think that having `children()`  returning `Iterable<Tree>` in the `JavaTree` is stupid... we should have go for a `List<Tree>` from day one, or a least a `Collection`. It does not make sense to use an Iterable anymore. If you are searching for something else to improve, go for it!

Reply:
It looks like there are cases where we use custom implementation of Iterable. So maybe we can update this in other PR

Reply:
Here it is https://github.com/SonarSource/sonar-java/pull/3345 :)